src/define_a_primitive_data_type.rs

```diff
@@ -0,0 +1,168 @@
+// http://rosettacode.org/wiki/Define_a_primitive_data_type
+// Implements a custom type named CustomInt.
+// This type only implements a subset of all traits within std::ops.
+
+mod custom_int {
+    use std::ops;
+
+    #[derive(Eq, PartialEq, Ord, PartialOrd, Debug, Copy, Clone)]
+    pub struct CustomInt {
+        value: u8,
+    }
+
+    // constructor function
+    pub fn custom_int(v: u8) -> CustomInt {
```

Comment:
I think this should be an associated function of `CustomInt` returning a `Result`.